okta-hosted-login/src/Navbar.jsx

```diff
@@ -13,13 +13,72 @@
 import { useOktaAuth } from '@okta/okta-react';
 import React from 'react';
 import { Link } from 'react-router-dom';
-import { Container, Icon, Image, Menu } from 'semantic-ui-react';
+import { Container, Icon, Image, Menu, Modal } from 'semantic-ui-react';
+import config from './config';
 
 const Navbar = () => {
+  const [errorModalOpen, setErrorModalOpen] = React.useState(false);
   const { authState, oktaAuth } = useOktaAuth();
 
+  // Note: Can't distinguish CORS error from other network errors
+  const isCorsError = (err) => (err.name === 'AuthApiError' && !err.errorCode && err.xhr.message === 'Failed to fetch');
+
+  // Build URL to "Trusted Origins" page in Admin panel
+  const { issuer } = config.oidc;
+  const baseUrl = issuer.split('/oauth2')[0];
+  const hostParts = new URL(baseUrl).host.split('.');
+  hostParts[0] += '-admin';
+  const adminHost = hostParts.join('.');
+  const corsAdminUrl = `https://${adminHost}/admin/access/api/trusted_origins`;
+
+  // URL to guide for enabling CORS
+  const guideUrl = 'https://developer.okta.com/docs/guides/enable-cors/granting-cors/';
+
   const login = async () => oktaAuth.signInWithRedirect();
-  const logout = async () => oktaAuth.signOut();
+
+  const logout = async () => {
+    try {
+      await oktaAuth.signOut();
+    } catch (err) {
+      if (isCorsError(err)) {
+        setErrorModalOpen(true);
+      } else {
+        throw err;
+      }
+    }
+  };
+
+  // CORS error modal
+  const errorModal = (
+    <Modal
+      onClose={() => setErrorModalOpen(false)}
+      open={errorModalOpen}
+      closeIcon
+    >
+      <Modal.Header>Network Error</Modal.Header>
+      <Modal.Content>
+        <Modal.Description>
+          <p>Seems like logout API call resulted with CORS error.</p>
+          <p>
+            You need to add your origin
```

Comment:
Since we don't know for certain this is a CORS error, let's add some qualification: You **may** need to add your origin.